### ISSUE TITLE
Library might not be working properly when debugging

### ISSUE THREAD

**Issue by Bornlex** [scastlara/minineedle#19]
When debugging using Pycharm, the objects in the current context are displayed in the debugger window, which calls their __str__ method.

The thing is the __str__ method itself calls the align method, filling the _alseq1 and _alseq2 attributes.

So when we call the self.align manually, both attributes are filled a second time, which creates a bug.

We could make sure the attributes are empty by assigning them to empty lists at the beginning of the _trace_back_alignment method:
`def _trace_back_alignment(self, irow: int, jcol: int) -> None:
        self._alseq1, self._alseq2 = [], []
        while True:
            if self._pmatrix[irow][jcol] == "diag":
                self._alseq1.append(self.seq1[jcol - 1])
                self._alseq2.append(self.seq2[irow - 1])
                if self.seq1[jcol - 1] == self.seq2[irow - 1]:
                    self._identity += 1
                irow -= 1
                jcol -= 1
            elif self._pmatrix[irow][jcol] == "up":
                self._alseq1.append(Gap(self._gap_character))
                self._alseq2.append(self.seq2[irow - 1])
                irow -= 1
            elif self._pmatrix[irow][jcol] == "left":
                self._alseq1.append(self.seq1[jcol - 1])
                self._alseq2.append(Gap(self._gap_character))
                jcol -= 1
            else:
                break
        self._alseq1 = list(reversed(self._alseq1))
        self._alseq2 = list(reversed(self._alseq2))
        self._identity = (self._identity / len(self._alseq1)) * 100`

**Comment by scastlara** [comment]
Hey! Thank you for commenting. I've never used Pycharm, so unfortunately I've never encountered that issue.

In reality, what you found out is a fundamental issue with the design that I made many years ago when writing this. Ideally, the aligner instance should not hold the state of the alignment, but return an instance of some kind of alignment class. Having said that, I am not interested in changing anything about this: changing it will just annoy the odd user.

So your fix looks fine to me. Feel free to make a PR if you want!

**Comment by Bornlex** [comment]
Yes exactly. I don't think you want to perform the align operation when calling the str method, which should be pure rendering.

So maybe a better way of doing it would be to:
- in the str method: just display empty sequence if the align has not be called yet
- in the align method: return the sequences already filled if the align method has been called already

Would it work for you?

I'm going to create the PR.

**Comment by Bornlex** [comment]
Actually I cannot push the branch I created with the fix

**Comment by scastlara** [comment]
> Actually I cannot push the branch I created with the fix. I cannot create a PR if the branch is only local

You would not be able to push to this repository. Instead you should create a fork of the repo, push a branch to it, and then create a pull request to this repository.

Here is a guide on how to do it: https://jarv.is/notes/how-to-pull-request-fork-github/
